common/djangoapps/third_party_auth/strategy.py

```diff
@@ -32,3 +32,21 @@ def setting(self, name, default=None, backend=None):
         # At this point, we know 'name' is not set in a [OAuth2|SAML]ProviderConfig row.
         # It's probably a global Django setting like 'FIELDS_STORED_IN_SESSION':
         return super(ConfigurationModelStrategy, self).setting(name, default, backend)
+
+    def create_user(self, *args, **kwargs):
+        if kwargs.get('autoprovision', False):
+            from student.views import create_account_with_params
+
+            user_fields = dict(kwargs)
+            user_fields['name'] = user_fields.get('fullname', "Unknown")  # needs to be >2 chars to pass validation
+            user_fields['honor_code'] = True
+            user_fields['terms_of_service'] = True
+
+            result = create_account_with_params(self.request, user_fields)
+
+        else:
+            result = super(ConfigurationModelStrategy, self).create_user(*args, **kwargs)
```

Comment:
Same comment as before - I think we should actually be raising an exception here. The pipeline flow should never get to this point.